### PR TITLE
Benchmark internal STARDIS functions

### PR DESCRIPTION
**Pull request by light2802**
### :pencil: Description

**Type:** :beetle: `bugfix` | :rocket: `feature` | :vertical_traffic_light: `testing` :roller_coaster: `infrastructure`

This PR adds workflow to benchmark internal stardis functions like raytrace

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by light2802** [comment]
This PR is based on top of PR #85 . So lets merge that first

**Comment by jvshields** [comment]
> This PR is based on top of PR #85 . So lets merge that first

Sorry, do you mean 74? 85 is this PR.

**Comment by light2802** [comment]
> > This PR is based on top of PR #85 . So lets merge that first
> 
> Sorry, do you mean 74? 85 is this PR.

Sorry , yes PR #74

File: benchmarks/run_stardis.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        base_dir = os.path.abspath(os.path.dirname(__file__))
+        self.config = os.path.join(base_dir, "benchmark_config.yml")
+        self.tracing_lambdas = np.arange(6540, 6590, 0.01) * u.Angstrom
+        os.chdir(base_dir)
+
+    def time_run_stardis(self):
+        run_stardis(self.config, self.tracing_lambdas)
+
+
+class BenchmarkRaytrace:
```
[the remark this answers is not in the document]
Comment (light2802) [reply benchmarks/run_stardis.py:37]: Are you suggesting an implementation as follows
`
class BenchmarkStardis:
    def setup:
        # Run run_stardis once and save generated variables
    def time_foo1:
         ....
    def time_foo2:
         .....
`

**Review by jvshields — DISMISSED** [review]

**Comment by atharva-2001** [comment]
I think this needs a rebase @light2802

**Review by jvshields — APPROVED** [review]

**Review by atharva-2001 — APPROVED** [review]

**Pull request merged**